File: Topics/Array/MoveZeroes.cpp
#include<iostream>
#include<vector>

using namespace std;

void MoveZeroes(vector<int> &arr , int n){
    int a = 0;

    for(int  i = 0 ; i<n-1 ; i++){
        if(arr[i] == 0){
            a = i ;
            break;
        }
    }

    cout<<"Xeroes Index : "<<a <<endl;

    for(int i = 0+a; i< n ; i++){
        if(arr[i] != 0){
            swap(arr[a++] , arr[i]);
        }
    }


    for(int row: arr){
        cout<<row << " ";
    }

}

int main(){
    vector<int> arr = {2 ,0, 4, 1, 3, 0, 28};

    MoveZeroes(arr ,  arr.size());

    return 0;
}

File: Array/MoveZeroes.java
package Array;

public class MoveZeroes {
    public static void main(String[] args) {
        int arr[] = {0 , 1 , 0 , 2 , 0 , 3 , 4 , 0 , 0 ,0 };
        int n =  arr.length ;
        int a =0;
        for(int i =0 ; i< n ;i++){
            if(arr[i] != 0) arr[a++] = arr[i] ;
        }

        while(a  != n){
            arr[a] = 0;
            a++;
        }

        for(int i =0; i< n ;i++){
            System.out.print(arr[i] + " ");
        }
    }
}
